src/test/java/uk/gov/digital/ho/pttg/ArchiveServiceTest.java

```diff
@@ -0,0 +1,216 @@
+        } catch(Exception e) {
+            // not testing the exception handling here
+        }
+
+
+        verify(mockAppender).doAppend(loggingEventArgumentCaptor.capture());
+        ILoggingEvent loggingEvent = loggingEventArgumentCaptor.getValue();
+        assertThat((loggingEvent.getArgumentArray()[0]))
+                .isEqualTo(new ObjectAppendingMarker("event_id", PTTG_AUDIT_ARCHIVE_FAILURE));
+    }
+
+    @Test
+    public void archiveResult_today_getsFullDay() {
+        LocalDateTime expectedStart = LocalDate.now().atStartOfDay();
+        LocalDateTime expectedEnd = LocalDate.now().plusDays(1).atStartOfDay();
+        when(mockRepository.findArchivedResults(expectedStart, expectedEnd)).thenReturn(Collections.emptyList());
+
+        archiveService.archiveResult(LocalDate.now(), "PASS");
+
+        verify(mockRepository).save(captorAuditEntry.capture());
+
+        AuditEntry actualAuditEntry = captorAuditEntry.getValue();
+        assertThat(actualAuditEntry.getTimestamp().toLocalDate()).isEqualTo(LocalDate.now());
+    }
+
+    @Test
+    public void archiveResult_newResult_resultSaved() {
+        when(mockRepository.findArchivedResults(any(LocalDateTime.class), any(LocalDateTime.class))).thenReturn(Collections.emptyList());
+
+        archiveService.archiveResult(LocalDate.now(), "PASS");
+
+        verify(mockRepository).save(captorAuditEntry.capture());
+
+        AuditEntry actualAuditEntry = captorAuditEntry.getValue();
+        JSONAssert.assertEquals("{\"results\": {\"PASS\": 1}}", actualAuditEntry.getDetail(), false);
+    }
+
+    @Test
+    public void archiveResult_existingArchiveNewResult_resultSaved() {
+        AuditEntry existingArchive = auditEntry(LocalDate.now(), "{\"results\": { \"FAIL\": 1}}");
+        when(mockRepository.findArchivedResults(any(LocalDateTime.class), any(LocalDateTime.class))).thenReturn(Collections.singletonList(existingArchive));
+
+        archiveService.archiveResult(LocalDate.now(), "PASS");
+
+        verify(mockRepository).save(captorAuditEntry.capture());
+
+        AuditEntry actualAuditEntry = captorAuditEntry.getValue();
+        JSONAssert.assertEquals("{\"results\": {\"PASS\": 1, \"FAIL\": 1}}", actualAuditEntry.getDetail(), false);
+    }
+
+    @Test
+    public void archiveResult_existingArchiveExistingResult_resultAppended() {
+        AuditEntry existingArchive = auditEntry(LocalDate.now(), "{\"results\": { \"PASS\": 1}}");
+        when(mockRepository.findArchivedResults(any(LocalDateTime.class), any(LocalDateTime.class))).thenReturn(Collections.singletonList(existingArchive));
+
+        archiveService.archiveResult(LocalDate.now(), "PASS");
+
+        verify(mockRepository).save(captorAuditEntry.capture());
+
+        AuditEntry actualAuditEntry = captorAuditEntry.getValue();
+        JSONAssert.assertEquals("{\"results\": {\"PASS\": 2}}", actualAuditEntry.getDetail(), false);
+    }
+
+    @Test(expected = ArchiveException.class)
```

Comment:
As this exception comes from a few lines in the class, have you considered assertJ's assertThatThrownBy(ThrowingCallable) or similar? That way you can assert on the message and check it's being thrown for the right reason.